### PR TITLE
Improve look of our forms

### LINKED ISSUE #3560: Announcements should be more visible
The announcements look like just another bookmark bar in the Browser. Since Announcements are more less important information for the user they should be more visible. 

## Possible Solution
- Colorize
- Maybe something like that: https://css-tricks.com/examples/PopFromTopMessage/

## Steps to Reproduce (for bugs)

1. Create an announcement
2. Ask your colleagues if they have noticed new the annoucment ;)

### PR DESCRIPTION
**Pull request by theFeu**
refs #3416 
fixes #3560

### CODE REVIEW

**Comment by theFeu** [comment]
What's still to do here is to go through our current forms and check if the new styles apply properly.

Add classes to the inputs, so we don't have to use the tags in the CSS and then adapt that.

File: public/css/icinga/forms.less [hunk 1]
```diff
@@ -1,618 +1,231 @@
...
+/**********************************************************************************************************************
+                                             specific input styles
+***********************************************************************************************************************/
 
-  &:focus {
-    background-color: @body-bg-color;
-    background-image: url('../img/icons/search_icinga_blue.png') !important;
-  }
+.control-group textarea {
+  background-image: url(../static/img?file=textarea-corner-2x.png&module_name=reporting);
```
Comment (lippserd) [inline public/css/icinga/forms.less]: The URL is wrong.

File: public/css/icinga/forms.less [hunk 2]
```diff
@@ -1,618 +1,231 @@
...
-  margin-left: 0.5em;
+.form-controls input:not(:last-child),
+.form-controls button:not(:last-child) {
+  margin-right: 1em;
 }
 
-button:hover .icon-cancel {
-  color: @color-critical;
+.control-group select {
+  background-image: url(../static/img?file=select-icon-2x.png&module_name=reporting);
```
Comment (lippserd) [inline public/css/icinga/forms.less]: The URL is wrong.

File: public/css/icinga/mixins.less [hunk 3]
```diff
@@ -6,26 +6,34 @@
           box-shadow: @arguments;
 }
 
-.button(@background-color: @body-bg-color, @color: @icinga-blue) {
+.button(@border-font-color: @icinga-blue, @color-dark: @icinga-blue-dark, @background-color: @body-bg-color) {
```
Comment (lippserd) [inline public/css/icinga/mixins.less]: Please don't change this mixin because we're not the only ones using it.

File: public/css/icinga/forms.less (hunk 1 above)
Comment (theFeu) [reply public/css/icinga/forms.less]: fixed in c2b886598579c3293727954afff8f8e383cdd73e

File: public/css/icinga/forms.less (hunk 2 above)
Comment (theFeu) [reply public/css/icinga/forms.less]: fixed in c2b886598579c3293727954afff8f8e383cdd73e

**Comment by theFeu** [comment]
> **Current Forms:**
> * Think about auto-submit spinner
> * Think about control-info

@lippserd 
We might need to discuss this before I change something up :)

File: public/css/icinga/forms.less [hunk 4]
```diff
@@ -78,6 +78,10 @@ form:not(.inline) {
   //todo(jem): maybe solve width with flexbox flex-growth: 1
 }
 
+.control-info {
+  width: 0;
```
Comment (theFeu) [inline public/css/icinga/forms.less]: @flourish86 is this too dirty?

**Comment by flourish86** [comment]
Here are some details to fix:

- [ ] Please add some opacity to the refresh indicators (~.4)
- [ ] Use the default cursor, when hovering disabled checkboxes
- [ ] Add negative margin to asterisks, if necessary wrap label text in another `<span>` to make sure asterisks won't get ellipsisized. This alignes the text borders
- [ ] Make sure to inset buttons by the label width to align with the inputs

**Comment by theFeu** [comment]
Maybe rethink the checkbox styles again and potentially solve the position with a 1 px margin or a transparent border for the :before

**Comment by theFeu** [comment]
Spinner width 0 makes them unusable in the module activation in safari
-> find different solution

**Comment by theFeu** [comment]
Replace the ellipsis for the form labels with a 2 line high box that just cuts off if the label is too long.

**Comment by theFeu** [comment]
Hover on spinner doesn't center the tooltip correctly

**Comment by theFeu** [comment]
input type number still has browser style

**Comment by theFeu** [comment]
should the select arrows be icingablue?

**Comment by theFeu** [comment]
form description has no styling yet - how do we want to display this?

**Comment by theFeu** [comment]
how do we display form validation / info messages?

**Comment by flourish86** [comment]
For Checkbox `:focus` state the outline isn't rounded. Please remove the outline for checkboxes only and use `box-shadow` instead.
- [ ] Remove outline for 'input[type="checkbox"]:focus` and replicate effect using `box-shadow`

**Comment by theFeu** [comment]
- [ ]  make checkboxes IE / Edge compatible

**Comment by theFeu** [comment]
So, I just found an old pen I made where I attempted to improve the handling of the role form
https://codepen.io/the_Feu/pen/zEeLdv

If you could have a look at it and tell me what you think @flourish86 that'd be neat!

**Comment by theFeu** [comment]
Where should the tooltips be for checkboxes?

Right next to them?
image

or

In a col with the others?
image

**Comment by theFeu** [comment]
- [ ] form description has no styling yet - how do we want to display this?

My first idea was to have rows with light gray background looking like this:
desktop:
image

mobile:
image

**Comment by theFeu** [comment]
- [ ] Should the select arrows be icinga blue?
Since it is an interaction and we want to let the user know this is clickable I would say: yes?

**Comment by flourish86** [comment]
> [x] Should the select arrows be icinga blue?
> Since it is an interaction and we want to let the user know this is clickable I would say: yes?

Yes, pls leave them as they are.

**Comment by theFeu** [comment]
- [ ] Add style for disabled checkboxes

**Comment by theFeu** [comment]
- [ ] style form errors

put the error message under the input field

**Comment by theFeu** [comment]
Still has no fluent layout switch - when window is pulled smaller there is a period of awkwardness...

image

**Comment by flourish86** [comment]
A side note: line-height is somehow odd on `select` elements, especially cross-browser wise.

I'm trying to fix this at the moment, with a reasonable amount of style code.

**Comment by flourish86** [comment]
I worked it all through and it seems like it's pretty done now.

There are still tiny imperfections with the filter editor. Nothing critical, though, so I think the current state is reasonably well in general and appropriate for merge. We should have the filter editor overworked in a separate issue.

A Review would be nice @lippserd.

**Comment by lippserd** [comment]
@flourish86 I had a look at the changes and noticed the following:
* Form controls have a submit animation
* The search bar in the menu has not the same height as the other menu entries. Though that's not new I guess. Should we change that? They have the same height when the sidebar is collapsed. Should we show the "search clear" button as well when the sidebar is collapsed?
* The filter editor has some style issues. See screenshot [1]
* The cancel button for the bottom notifications is kind of lost. See screenshot [2]
* The same goes for announcements. See screenshot [3]

[1]
Screenshot 2019-07-03 at 10 38 47

[2]
Screenshot 2019-07-04 at 11 40 32

[3]
Screenshot 2019-07-04 at 13 13 01

**Comment by flourish86** [comment]
Can't help, but having the nav search input being as tall as the nav items seems somehow odd.

Screenshot 2019-07-04 at 17 46 11

**Comment by flourish86** [comment]
> Form controls have a submit animation

Should be fixed by 33fbe30

> The search bar in the menu has not the same height as the other menu entries. Though that's not new I guess. Should we change that? They have the same height when the sidebar is collapsed. Should we show the "search clear" button as well when the sidebar is collapsed?

As pointed out above, I wouldn't change as it looks more balanced now.

> The filter editor has some style issues. See screenshot [1]

Should be fixed by 0cd54c8

> The cancel button for the bottom notifications is kind of lost. See screenshot [2]

Fixed by 620ae0a

> The same goes for announcements. See screenshot [3]

Fixed by 456c48f

**Comment by flourish86** [comment]
@nilmerg Still to do:

1) The `<em>`s wrapping the highlighted elements in the labels are still missing.
2) The permission and restriction subheading is contained in a `.control-group`, which also contains a `.control-label-group`. I couldn't find a way to make the headings left-aligned this way. Can you please remove the `.control-label-group`?

**Comment by lippserd** [comment]
@flourish86 Could please fix the following issues:

* The search input height is still too low in my opinion
* The bottom notifications look displaced

localhost_8080_icingaweb2_dashboard

**Comment by lippserd** [comment]
Plus, if the side bar is collapsed and you hover over the search icon, the changed background does not fill the full tile height.

**Comment by flourish86** [comment]
@lippserd: fixed by 4b88dd8

**Comment by lippserd** [comment]
Thx! Any chance to fix the reset button flickering upon auto-refresh of the menu?

**Comment by flourish86** [comment]
> Thx! Any chance to fix the reset button flickering upon auto-refresh of the menu?

Commit 9b5b604 removes fade effect. It also hides the button, when the input is empty (not `:valid`)

**Comment by lippserd** [comment]
@flourish86 Did you fix the bottom notifications?

**Comment by flourish86** [comment]
Now ccd4d4d

File: public/css/icinga/widgets.less [hunk 5]
```diff
@@ -66,6 +68,10 @@
 
   position: relative;
 
+  p {
+    margin-bottom: 0;
```
Comment (nilmerg) [inline public/css/icinga/widgets.less]: @flourish86 There's no bottom margin even without this. Why is this necessary?

Comment (nilmerg) [reply public/css/icinga/widgets.less]: Never mind, I bet you've fixed the bottom notifications without knowing I've done this already with ef7bdf807ff75533e91c82ecf47758809bb12e3b

Comment (lippserd) [reply public/css/icinga/widgets.less]: I guess that's true 😆

**Comment by lippserd** [comment]
I can't expand the modules in the role form.

**Comment by nilmerg** [comment]
Should be fixed now :sweat:

**Comment by lippserd** [comment]
Looks good!

**Comment by sinyb** [comment]
With a little help from Firefox debugging, I have changed a few files (relative to icingaweb2):
modules/monitoring/application/views/scripts/list/hosts.phtml
modules/monitoring/application/views/scripts/list/services.phtml
modules/monitoring/public/css/module.less
modules/monitoring/public/css/tables.less
public/css/icinga/base.less
and managed to squeeze much more useful info on a single screen, as can be seen down in screenshots. I needed this because out of 2500 hosts, there are always at least 20 with problems, and I want to see them all at once.
Now, I wonder if it could be possible to pack this into a Theme, so I don't have to change everything again after an update?
I suppose that I can override .less files via public/css/themes  (will try that next time I have some spare time), but what about .phtml? I needed to delete some div's, and reorder a bit of php/html code (I can attach diffs if anybody is interested). This is what I was writing about in issue #3416  ...

Before
Screenshot_20200127_213203

After (not on the same system):
Screenshot_20200127_213208

**Comment by nilmerg** [comment]
@sinyb Please head over to our community forums https://community.icinga.com where you'll certainly get more help sooner.
